x-pack/legacy/plugins/maps/public/shared/layers/vector_layer.js

```diff
@@ -95,6 +95,28 @@ export class VectorLayer extends AbstractLayer {
     });
   }
 
+  isDataLoaded(zoom) {
+    if (!this.isVisible() || !this.showAtZoomLevel(zoom)) {
```

Comment:
I would remove this check and move the evaluation of visibility to the selector. 

Alternatively, rename the method to something else: `isLayerVisibleAndReady` or whatever ...